pkg/logql/log/distinct_test.go

```diff
@@ -0,0 +1,61 @@
+package log
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/pkg/logqlmodel"
+)
+
+func Test_DistinctFilter(t *testing.T) {
+
+	c := struct {
+		name          string
+		label         []string
+		lbs           labels.Labels
+		input         []string
+		expectedCount int
+		expectedLines []string
+	}{
+		name:  "distinct test",
+		label: []string{"id", "time"},
+		lbs: labels.Labels{
+			{Name: logqlmodel.ErrorLabel, Value: errJSON},
+			{Name: "status", Value: "200"},
+			{Name: "method", Value: "POST"},
+		},
+		input: []string{
+			`{"event": "access", "id": "1", "time": "1"}`,
+			`{"event": "access", "id": "1", "time": "2"}`,
+			`{"event": "access", "id": "2", "time": "3"}`,
+			`{"event": "access", "id": "2", "time": "4"}`,
+			`{"event": "access", "id": "1", "time": "5"}`,
+		},
+		expectedCount: 2,
+		expectedLines: []string{
+			`{"event": "access", "id": "1", "time": "1"}`,
+			`{"event": "access", "id": "2", "time": "3"}`,
+		},
+	}
+
+	distinctFilter, err := NewDistinctFilter(c.label)
+	require.NoError(t, err)
+
+	total := 0
+	passLines := make([]string, 0)
+	for _, line := range c.input {
+		b := NewBaseLabelsBuilder().ForLabels(c.lbs, c.lbs.Hash())
+		NewJSONParser().Process(1, []byte(line), b)
+		_, pass := distinctFilter.Process(1, []byte(line), b)
+		if pass {
+			total++
+			passLines = append(passLines, line)
```

Comment:
does `pass` here mean `filter out`? if so, WDYT of naming `total` as `filteredLines` or `droppedLines` or something similar? I got a little confused if `pass` is "pass this line forwards" (i.e: do not drop it) or "pass/skip this line"

Reply:
pass is "pass this line forwards"

I searched and most of the test cases use `ok`

![image](https://user-images.githubusercontent.com/9583245/233525273-19724434-101b-46ab-a4fc-2353d0c3445b.png)

Reply:
got it, thank you. "ok" is good for the consistency", "keep" or something similar is good for clarity; whatever you decide I'll be happy with :smile:

Reply:
👌, ☑️ "ok"

Reply:
done